src/slave/task_status_update_manager.cpp

```diff
@@ -881,11 +881,21 @@ void TaskStatusUpdateStream::_handle(
     // Add it to the pending updates queue.
     pending.push(update);
   } else {
-    // Record this ACK.
-    acknowledged.insert(id::UUID::fromBytes(update.uuid()).get());
+    StatusUpdate& pendingUpdate = pending.front();
 
-    // Remove the corresponding update from the pending queue.
-    pending.pop();
+    if (pendingUpdate.uuid() == update.uuid()) {
```

Comment:
Can we use the _operator==()_ on the _id::UUID_ to compare the two fields instead?